src/sentry/api/endpoints/organization_integration_repository_project_path_configs.py

```diff
@@ -90,38 +107,102 @@ def update(self, instance, validated_data):
         return self.instance
 
 
+class NullableOrganizationIntegrationMixin:
+    def get_organization_integration(self, organization, integration_id):
+        try:
+            return OrganizationIntegration.objects.get(
+                integration_id=integration_id,
+                organization=organization,
+            )
+        except OrganizationIntegration.DoesNotExist:
+            raise Http404
+
+    def get_project(self, organization, project_id):
+        try:
+            return Project.objects.filter(organization=organization, id=project_id).get()
+
+        except Project.DoesNotExist:
+            raise Http404
+
+
 class OrganizationIntegrationRepositoryProjectPathConfigEndpoint(
-    OrganizationIntegrationBaseEndpoint
+    OrganizationEndpoint, NullableOrganizationIntegrationMixin
 ):
     permission_classes = (OrganizationIntegrationsPermission,)
 
-    def get(self, request, organization, integration_id):
+    def get(self, request, organization):
         """
-        Get the list of repository project path configs in an integration
+        Get the list of repository project path configs
+
+        :pparam string organization_slug: the slug of the organization the
+                                          team should be created for.
+        :queryparam int integrationId: the optional integration id.
+        :queryparam int projectId: the optional project id.
+        :auth: required
         """
-        org_integration = self.get_organization_integration(organization, integration_id)
+        integration_id = request.GET.get("integrationId")
+        project_id = request.GET.get("projectId")
 
-        # front end handles ordering
-        repository_project_path_configs = RepositoryProjectPathConfig.objects.filter(
-            organization_integration=org_integration
-        )
+        if not integration_id and not project_id:
```

Comment:
@NisanthanNanthakumar I would recommend using a `CamelSnakeSerializer` here and then implement this logic in the `validate` function

Reply:
@scefali we're already using `CamelSnakeModelSerializer`. Can definitely move this to the validate function.

Reply:
@NisanthanNanthakumar cool. In general, we should seek to raise all `400` errors in the serializers.

Reply:
@scefali actually this is the GET request and we don't instantiate the serializer. The `is_valid` method will run all the validators, like validating `repository_id` which will not exist for the GET. 

I think it's ok to keep the query param validation logic here. We have done this before: https://github.com/getsentry/sentry/blob/master/src/sentry/api/endpoints/organization_teams.py#L80

Reply:
@NisanthanNanthakumar I'll admit maybe I don't know the best practice here but I'd like to see what folks say on discuss-backend abouut it

Reply:
Looks like Dan thinks it's OK either way so I'll resolve this conversation